### DISCUSSION TITLE
Rate limit documentation for NPM Registry API

### DISCUSSION THREAD

**Discussion by Ensar025** [npm/feedback#658] [Documentation] [answered]
Is there any recent documentation available defining the rate limits for the NPM Registry API? The only one I was able to find is an [NPM blog from 2017](https://blog.npmjs.org/post/164799520460/api-rate-limiting-rolling-out.html), is this still relevant? Thanks in advance.

**Comment by TimChermin** [comment]
I was not able to find anything recent either.

**Comment by PengBAI** [comment]
anybody could give more informations please ?

**Comment by MylesBorins** [comment, marked as answer]
We are currently determining a strategy for documenting the npm API surface. API docs are a pre-requisite for us to document our rate limiting per API

**Reply by agschrei** [reply]
@MylesBorins is there a publicly visible task for this process or is this something you will exclusively handle internally?

We have a number of tools in deployment that query the Registry API for package metadata and I'd like to be a good citizen and enforce client-side rate-limiting based on the limits you put forth.

**Reply by jeremykenedynv5** [reply]
When building out a team in npmjs.org, rate limiting gets in the way of invites.... and there is zero documentation on what those limits are or when I can expect to be able to continue... Would you be able to advise the rate limit? Thank you :)

**Reply by jtlapp** [reply]
I'm currently getting the error `429 Too Many Requests` on `npm publish`. Have I been rate limited? When can I resume?

**Reply by jonstuebe** [reply]
I'm getting the same thing and can't find any docs on how the rate limiting get's applied. I understand this is an old discussion, but also no updates in almost a year?

**Reply by Jey-Cee** [reply]
The same here. [Status](https://status.npmjs.org/) shows no problems.

**Reply by lehni** [reply]
Same here!

**Reply by jonstuebe** [reply]
Looks like the issue has been resolved

**Reply by jtlapp** [reply]
I sent a support request, and they responded that they've fixed the issue.

**Reply by barrymichaeldoyle** [reply]
Are there any updates to this? I'm building a CLI tool to check whether dependencies in my package.json are out of date and I want to ensure I won't hit a rate limiting issue...

**Comment by adam-browning** [comment]
Is there any fresh information on NPM rate limiting? 
Terms & conditions do not mention anything on the subject

**Comment by cwtuan** [comment]
How to use private token to access npm api for larger rate limit?
